execution-core/src/transfer/phoenix.rs

```diff
@@ -273,6 +273,13 @@ impl Transaction {
         Self { payload, proof }
     }
 
+    /// Replaces the inner `proof` bytes for a given `proof`.
+    /// Note: This method is likely to invalidate the transaction and should
+    /// only be used with care.
+    pub fn replace_proof(&mut self, proof: Vec<u8>) {
```

Comment:
This is basically a setter, so idiomatically I would name this method `set_proof`.